src/DocumentFormat.OpenXml/Packaging/OpenXmlPartContainer.cs

```diff
@@ -1703,45 +1703,69 @@ internal OpenXmlPart AddSubPartFromOtherPackage(OpenXmlPart part, bool keepIdAnd
                 }
 
-                var updatedParts = new Dictionary<DataPart, DataPart>();
-                foreach (var item in dataPartsDictionary)
+                UpdateDataParts(dataPartsDictionary);
+
+                // then create data part reference relationship
+                foreach (var dataPartReferenceRelationship in part.DataPartReferenceRelationships)
                 {
-                    if (item.Value is null)
+                    if (dataPartsDictionary[dataPartReferenceRelationship.DataPart] is MediaDataPart newDataPart)
                     {
-                        var dataPart = item.Key;
-
-                        var newDataPart = new MediaDataPart(InternalOpenXmlPackage, dataPart.ContentType, dataPart.Uri);
+                        var newDataPartReference = DataPartReferenceRelationship.Create(child, newDataPart, dataPartReferenceRelationship.RelationshipType, dataPartReferenceRelationship.Id);
+                        child.AddDataPartReferenceRelationship(newDataPartReference);
+                    }
+                }
 
-                        // copy the stream
-                        using (var stream = dataPart.GetStream())
-                        {
-                            newDataPart.FeedData(stream);
-                        }
+                return child;
+            }
+        }
 
-                        InternalOpenXmlPackage.AddDataPartToList(newDataPart);
+        // Updates of existing dictionary keys during enumeration are only allowed after net 5.0.
+        // Before that we need a temporary dictionary to store the updated values for the keys.
+#if NET50_OR_LATER
+        private void UpdateDataParts(Dictionary<DataPart, DataPart?> dataPartsDictionary) {
+            foreach (var (key, value) in dataPartsDictionary)
+            {
+                if (value is null)
+                {
+                    var newDataPart = new MediaDataPart(InternalOpenXmlPackage, key.ContentType, key.Uri);
 
-                        updatedParts.Add(dataPart, newDataPart);
+                    // copy the stream
+                    using (var stream = key.GetStream())
+                    {
+                        newDataPart.FeedData(stream);
                     }
-                }
 
-                foreach (var item in updatedParts)
-                {
-                    dataPartsDictionary[item.Key] = item.Value;
+                    InternalOpenXmlPackage.AddDataPartToList(newDataPart);
+                    dataPartsDictionary[key] = newDataPart;
                 }
-
-                // then create data part reference relationship
-                foreach (var dataPartReferenceRelationship in part.DataPartReferenceRelationships)
+            }
+        }
+#else
+        private void UpdateDataParts(Dictionary<DataPart, DataPart?> dataPartsDictionary) {
+            var updatedParts = new Dictionary<DataPart, DataPart>();
+            foreach (var (key, value) in dataPartsDictionary)
```

Comment:
Failing because we don't have access to ValueTuple generally in the project (only in targets that include it, ie .NET 5,.NET Standard 2.0)

Reply:
Sorry, forgot to check the build wiht an older .NET standard. Should be fixed now